### PR TITLE
Add robin.js

### PR DESCRIPTION
**Pull request by rickhanlonii**

### CODE REVIEW

File: robin.js [hunk 1]
```diff
@@ -0,0 +1,852 @@
...
+      vote: function (e) {
+        this.room.isComplete() ? this.addSystemMessage("voting is complete") : this.room.postVote(e.toUpperCase())
+      }
+    },
+    quitWidgetEvents: {
+      quit: function () {
+        this.addSystemMessage("leaving room..."), this.room.postLeaveRoom()
+      }
+    },
+    chatCommands: {
```
Comment (rickhanlonii) [inline robin.js:677]: Here are all the chat commands.

File: robin.js [hunk 2]
```diff
@@ -0,0 +1,852 @@
...
+    var a = e + " | " + t;
+    this.name = "ApiError", this.message = a, this.displayName = e, this.displayMessage = t, this.field = n || "", this.source = s || "api"
+  }
+
+  var t = {
+    UNKNOWN_ERROR: e._("unknown error %(message)s"),
+    NO_TEXT: e._("we need something here"),
+    TOO_LONG: e._("this is too long (max: %(max_length)s)"),
+    TOO_SHORT: e._("this is too short (min: %(min_length)s)"),
+    SR_RULE_EXISTS: e._("A subreddit rule by that name already exists."),
```
Comment (rickhanlonii) [inline robin.js:188]: This is interesting. Can track this down to figure out how a subreddit gets created.

File: robin.js [hunk 3]
```diff
@@ -0,0 +1,852 @@
...
+      this.lastMessageFrom = null;
+      var n = e.find(".robin-message--message"), r = n.text(), i = [], s;
+      for (var o = 0; o < r.length; o++)s = t(t.parseHTML("<span></span>")), s.text(r[o]), r[o] !== " " ? s.addClass(this.CHAR_CLASS) : s.addClass(this.SPACE_CHAR_CLASS), s.css({"transition-delay": Math.floor(Math.random() * 1e3) + "ms"}), i.push(s);
+      return n.empty().append(i), setTimeout(function () {
+        e.addClass(this.JUICE_CLASS)
+      }.bind(this), 10), setTimeout(function () {
+        e.remove()
+      }.bind(this), 1500), !0
+    },
+    startJuicyPoppin: function () {
```
Comment (rickhanlonii) [inline robin.js:364]: This is triggered by `/clear` and causes the animation.

File: robin.js [hunk 4]
```diff
@@ -0,0 +1,852 @@
...
+    quitWidgetEvents: {
+      quit: function () {
+        this.addSystemMessage("leaving room..."), this.room.postLeaveRoom()
+      }
+    },
+    chatCommands: {
+      unknown: function (e) {
+        this.addSystemMessage('"/' + e + '" is not a command')
+      }, help: function () {
+        this.addSystemMessage("Welcome to Robin."), this.addSystemMessage("Be sure to use the buttons in the sidebar to vote on the future of the room before the polls are closed."), this.addSystemMessage("Non-votes and abstentions will be counted as votes to abandon."), this.addSystemMessage("We do hope you enjoy the discussion.")
```
Comment (rickhanlonii) [inline robin.js:681]: Non-votes are abandons.

File: robin.js (hunk 3 above)
Comment (reustle) [inline robin.js:364]: Would be cool to extract this as a lib

**Pull request merged**